### ISSUE TITLE
EOS workchain failing for CPA

### ISSUE THREAD

**Issue by JPchico** [JuDFTteam/aiida-kkr#69]
When submitting a EOS calculation with an alloy the workchain fails after the first voronoi calculation. 

The calculation fails when in the `kkr_scf_wc` it tries to get the `last_params_voronoi`
```
self.ctx.last_params = self.ctx.voronoi.outputs.last_params_voronoi

aiida.common.exceptions.NotExistentAttributeError: Node<134354> does not have an output with link label 'last_params_voronoi'
```

However, if one submits a single SCF calculation it works without problems, indicating that there is some sort of issue when passing data between the SCF calculation.

The `vorostart` calculation fails with exit code 203
```
type         kkr_startpot_wc
state        Finished [203] Voronoi calculation unsuccessful. Check inputs
```

The `VoronoiCalculation` fails when setting up the `RMTCORE`
```
 | [134390|VoronoiCalculation|on_except]: Traceback (most recent call last):
 |   File "/home/jonathan/Codes/AiiDA/aiida-core/aiida/engine/processes/calcjobs/tasks.py", line 85, in do_upload
 |     calc_info = process.presubmit(folder)
 |   File "/home/jonathan/Codes/AiiDA/aiida-core/aiida/engine/processes/calcjobs/calcjob.py", line 522, in presubmit
 |     calc_info = self.prepare_for_submission(folder)
 |   File "/home/jonathan/Codes/Forks/aiida-kkr/aiida_kkr/calculations/voro.py", line 178, in prepare_for_submission
 |     natom, nspin, newsosol, warnings_write_inputcard = generate_inputcard_from_structure(
 |   File "/home/jonathan/Codes/Forks/aiida-kkr/aiida_kkr/tools/common_workfunctions.py", line 735, in generate_inputcard_from_structure
 |     params.fill_keywords_to_inputfile(output=input_filename)
 |   File "/home/jonathan/Codes/Forks/masci-tools/masci_tools/io/kkr_params.py", line 1248, in fill_keywords_to_inputfile
 |     self._check_input_consistency()
 |   File "/home/jonathan/Codes/Forks/masci-tools/masci_tools/io/kkr_params.py", line 1204, in _check_input_consistency
 |     self._check_array_consistency()
 |   File "/home/jonathan/Codes/Forks/masci-tools/masci_tools/io/kkr_params.py", line 1153, in _check_array_consistency
 |     raise TypeError('Error: array input not consistent for key {}'.format(key))
 | TypeError: Error: array input not consistent for key <RMTCORE>
```

When looking at the input data one can see that `RMTCORE` seems to be well setup `'<RMTCORE>': [2.2906805904695],` (for a single site structure with two chemical species).

**Comment by JPchico** [comment]
Trying to reproduce the error using the `generate_inputcard_from_structure` function I see that the issue is indeed related to the `RMTCORE`. 
```
Warning: filling value of "<ZATOM>" with integer but expects float. Converting automatically and continue
check consistency: <RMTCORE> [2.2906805904695] (2,) (1,) False
```

**Comment by PhilippRue** [comment]
> Seems to be that for some reason it assumes that `RMTCORE` has a size of 2 instead of 1?

How does the input structure look like? I guess it has something to do with the way natyp/naez are found from the structure.

**Comment by JPchico** [comment]
@PhilippRue the structure is an fcc system with one atom per unit cell with two species, it is a 50, 50 binary.

**Comment by JPchico** [comment]
@PhilippRue I think it is related to this [issue](https://github.com/JuDFTteam/masci-tools/issues/50).  I had misunderstood what you wrote, I think that right now `RMTCORE` is using `natyp` 

https://github.com/JuDFTteam/masci-tools/blob/aa3f2ebb17999a114da6a8f4440e67df552e6546/masci_tools/io/kkr_params.py#L1182-L1192

Instead it should be `naez` right? So what is happening is that in the EOS workchain you set the `RMTCORE` in the input after the first calculation, and then as `natyp != naez` for alloys then you have an issue.

**Comment by PhilippRue** [comment]
> @PhilippRue I think it is related to this [issue](https://github.com/JuDFTteam/masci-tools/issues/50). I had misunderstood what you wrote, I think that right now `RMTCORE` is using `natyp`
> 
> https://github.com/JuDFTteam/masci-tools/blob/aa3f2ebb17999a114da6a8f4440e67df552e6546/masci_tools/io/kkr_params.py#L1182-L1192
> 
> Instead it should be `naez` right? So what is happening is that in the EOS workchain you set the `RMTCORE` in the input after the first calculation, and then as `natyp != naez` for alloys then you have an issue.

I was just about to write the same thing ;)

> I think the error in the end is that  in masci-tools the [RMTCORE input is expected to have the `natyp` dimension](https://github.com/JuDFTteam/masci-tools/blob/aa3f2ebb17999a114da6a8f4440e67df552e6546/masci_tools/io/kkr_params.py#L1191) (2 for the CPA structure) but is is only set with the `naez` (as it should be).
So changing the expected size for the RMTCORE input in masci-tools to `naez` instead of `natyp` should resolve this.

**Comment by JPchico** [comment]
I can make a quick PR if you want, it is just two lines that need to be changed =)

**Comment by JPchico** [comment]
Fixed via #62
